src/test/java/core/basesyntax/service/StorageServiceImplTest.java

```diff
@@ -0,0 +1,110 @@
+package core.basesyntax.service;
+
+import static core.basesyntax.db.Storage.storage;
+import static org.junit.Assert.assertEquals;
+
+import core.basesyntax.service.model.FruitTransaction;
+import core.basesyntax.service.operations.BalanceOperationHandler;
+import core.basesyntax.service.operations.OperationHandler;
+import core.basesyntax.service.operations.PurchaseOperationHandler;
+import core.basesyntax.service.operations.ReturnOperationHandler;
+import core.basesyntax.service.operations.SupplyOperationHandler;
+import core.basesyntax.strategy.OperationHandlerStrategyImpl;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.After;
+import org.junit.Test;
+
+public class StorageServiceImplTest {
+    private static final FruitTransaction NULL_OPERATION =
+            new FruitTransaction(null, "banana", 123);
+    private static final FruitTransaction NULL_FRUIT =
+            new FruitTransaction(FruitTransaction.TypeOperation.PURCHASE, null, 123);
+    private static final FruitTransaction EMPTY_FRUIT =
+            new FruitTransaction(FruitTransaction.TypeOperation.PURCHASE, "", 123);
+    private static final FruitTransaction NULL_QUANTITY =
+            new FruitTransaction(FruitTransaction.TypeOperation.SUPPLY, "apple", null);
+    private static final FruitTransaction TO_BIG_QUANTITY =
+            new FruitTransaction(FruitTransaction.TypeOperation.SUPPLY, "apple", Integer.MAX_VALUE);
+
+    private static final FruitTransaction WRIGHT_BALANCE =
+            new FruitTransaction(FruitTransaction.TypeOperation.BALANCE, "apple", 25);
+    private static final FruitTransaction WRIGHT_SUPPLY =
+            new FruitTransaction(FruitTransaction.TypeOperation.SUPPLY, "apple", 50);
+    private static final FruitTransaction WRIGHT_PURCHASE =
+            new FruitTransaction(FruitTransaction.TypeOperation.PURCHASE, "apple", 15);
+    private static final FruitTransaction WRIGHT_RETURN =
+            new FruitTransaction(FruitTransaction.TypeOperation.RETURN, "apple", 5);
+    private static StorageServiceImpl storageService;
+
+    public StorageServiceImplTest() {
+        Map<FruitTransaction.TypeOperation, OperationHandler> listOperations = new HashMap<>();
+        listOperations.put(FruitTransaction.TypeOperation.BALANCE, new BalanceOperationHandler());
+        listOperations.put(FruitTransaction.TypeOperation.SUPPLY, new SupplyOperationHandler());
+        listOperations.put(FruitTransaction.TypeOperation.PURCHASE, new PurchaseOperationHandler());
+        listOperations.put(FruitTransaction.TypeOperation.RETURN, new ReturnOperationHandler());
+        storageService = new StorageServiceImpl(new OperationHandlerStrategyImpl(listOperations));
+    }
+
+    @Test (expected = RuntimeException.class)
+    public void nullOperationItem_NotOK() {
```

Comment:
Remember about the names convention. Start with method-under-test name
```suggestion
    public void operation_nullOperationItem_NotOK() {
```